ipv8/attestation/trustchain/community.py

```diff
@@ -230,10 +284,6 @@ def process_half_block(self, blk, peer):
         self.logger.debug("Block validation result %s, %s, (%s)", validation[0], validation[1], blk)
         if validation[0] == ValidationResult.invalid:
             return
-        elif not self.persistence.contains(blk):
-            self.persistence.add_block(blk)
```

Comment:
Why was this removed?

Reply:
Because the block is already added to the database in the `validate_persist_block` method if it is valid. These two lines I removed here were duplicate and unnecessary.